server/service/devices.go

```diff
@@ -143,6 +145,15 @@ func getDeviceHostEndpoint(ctx context.Context, request interface{}, svc fleet.S
 		},
 	}
 
+	resp.DEPAssignedToFleet = ptr.Bool(false)
+	if ac.MDM.EnabledAndConfigured && license.IsPremium() {
+		hdep, err := svc.GetHostDEPAssignment(ctx, host)
+		if err != nil && !fleet.IsNotFound(err) {
+			return getDeviceHostResponse{Err: err}, nil
+		}
+		resp.DEPAssignedToFleet = ptr.Bool(hdep != nil && hdep.IsDEPAssignedToFleet())
```

Comment:
Nit, but as mentioned above, if `hdep.IsDEPAssignedToFleet` makes the `nil` check, you can just call `ptr.Bool(hdep.IsDEPAssignedToFleet())`.